beacon-chain/execution/engine_client.go

```diff
@@ -659,7 +673,60 @@ func (s *Service) ReconstructBlobSidecars(ctx context.Context, block interfaces.
 // ReconstructDataColumnSidecars reconstructs the verified data column sidecars for a given beacon block.
 // It retrieves the KZG commitments from the block body, fetches the associated blobs and cell proofs from the EL,
 // and constructs the corresponding verified read-only data column sidecars.
+// This method includes retry logic for getBlobsV2 calls and prevents duplicate concurrent calls.
 func (s *Service) ReconstructDataColumnSidecars(ctx context.Context, signedROBlock interfaces.ReadOnlySignedBeaconBlock, blockRoot [fieldparams.RootLength]byte) ([]blocks.VerifiedRODataColumn, error) {
+	// Check if there's already an in-flight reconstruction call for this block
+	if resultChanInterface, exists := s.activeReconstructCalls.Load(blockRoot); exists {
+		resultChan := resultChanInterface.(chan reconstructResult)
+		// Wait for the in-flight call to complete and return its result
+		select {
+		case result := <-resultChan:
+			return result.result, result.err
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	}
+
+	// Create a result channel for this reconstruction call
+	resultChan := make(chan reconstructResult, 1)
+
+	// Try to claim this reconstruction call
+	if existingChanInterface, loaded := s.activeReconstructCalls.LoadOrStore(blockRoot, resultChan); loaded {
+		existingChan := existingChanInterface.(chan reconstructResult)
+		// Another goroutine claimed it, wait for that result
+		select {
+		case result := <-existingChan:
+			return result.result, result.err
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	}
+
+	// We claimed the reconstruction call, ensure cleanup
+	defer s.activeReconstructCalls.Delete(blockRoot)
+	defer close(resultChan)
+
+	// Perform the actual reconstruction
+	result, err := s.reconstructDataColumnSidecarsOnce(ctx, signedROBlock, blockRoot)
+
+	// Notify all waiters of the result (including the error)
+	resultChan <- reconstructResult{result: result, err: err}
+
+	// Only start retry if we need more data and don't have an active retry
+	if (err != nil || len(result) == 0) && !s.hasActiveRetry(blockRoot) {
+		// Check if we already have all required data
+		if !s.isDataAlreadyAvailable(ctx, blockRoot, signedROBlock) {
+			// Start background retry goroutine
+			go s.retryReconstructDataColumnSidecars(ctx, signedROBlock, blockRoot)
```

Comment:
I think we don't need to define `s.reconstructDataColumnSidecarsOnce` AND `s.retryReconstructDataColumnSidecars`.
The same function could do the same. (First try to reconstruct, then if not available, then setup the ticker.)